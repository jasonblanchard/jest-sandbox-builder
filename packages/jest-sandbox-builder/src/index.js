import get from 'lodash.get';
import set from 'lodash.set';

// TODO: Split up this file.

const registry = {};
const stack = [];
let lastIt;

export function describe(name, fn) {
  if (stack[stack.length - 1]) {
    registry[stack[stack.length - 1]][name] = {};
  } else {
    registry[name] = {};
  }
  stack.push(name);
  fn();
  stack.pop();
}

export function it(name, fn) {
  set(registry, [...stack, name], null);
  lastIt = name;
  fn();
}

export function beforeEach(fn) {
  set(registry, [...stack, 'beforeEach'], null);
  lastIt = 'beforeEach';
  fn();
}

export function factory(factoryFn, renderFn) {
  const component = factoryFn();
  set(registry, [...stack, lastIt], {
    isLeaf: true,
    component
  });

  if (process.env.NODE_ENV === 'test') return renderFn(component);

  // TODO: Abstract and generalize
  return {
    findWhere: () => ({
      exists: () => {},
      simulate: () => {}
    })
  };
}

export function getRegistry() {
  return registry;
}

// TODO: Mock all Jest globals https://facebook.github.io/jest/docs/en/api.html
export function expect() {
  return {
    toMatchSnapshot: () => {},
    toEqual: () => {},
    toBeCalled: () => {}
  };
}

export const jest = {
  fn: () => {}
}
