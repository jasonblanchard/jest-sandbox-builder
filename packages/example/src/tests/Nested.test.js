import { mount } from 'enzyme';
import React from 'react';

import Nested from '../Nested';
import { factory } from 'jest-sandbox-builder';

function render(props) {
  return factory(() => {
    return <Nested {...props} />;
  }, mount);
}

describe('Nested', () => {
  it('renders', () => {
    const wrapper = render();
    expect(wrapper).toMatchSnapshot();
  });
});
