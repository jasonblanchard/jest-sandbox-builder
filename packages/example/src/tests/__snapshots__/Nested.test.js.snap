// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nested renders 1`] = `
<Nested>
  <div
    className="Nested"
  >
    This is a nested component 
    <button>
      click me
    </button>
  </div>
</Nested>
`;
